code/modules/religion/sparring/sparring_datum.dm

```diff
@@ -0,0 +1,293 @@
+
+/datum/sparring_match/proc/hook_signals(mob/living/carbon/human/sparring)
+	//weapon conditions
+	if(weapons_condition < ANY_WEAPON)
+		RegisterSignal(sparring, COMSIG_MOB_FIRED_GUN, .proc/gun_violation)
+		RegisterSignal(sparring, COMSIG_MOB_GRENADE_ARMED, .proc/grenade_violation)
+	if(weapons_condition < MELEE_ONLY)
+		RegisterSignal(sparring, COMSIG_MOB_ITEM_ATTACK, .proc/melee_violation)
+	//arena conditions
+	RegisterSignal(sparring, COMSIG_MOVABLE_MOVED, .proc/arena_violation)
+	//severe violations (insta violation win for other party) conditions
+	RegisterSignal(sparring, COMSIG_MOVABLE_TELEPORTED, .proc/teleport_violation)
+	//win conditions
+	RegisterSignal(sparring, COMSIG_MOB_STATCHANGE, .proc/check_for_victory)
+	//flub conditions
+	RegisterSignal(sparring, COMSIG_PARENT_ATTACKBY, .proc/outsider_interference)
+	RegisterSignal(sparring, COMSIG_ATOM_HULK_ATTACK, .proc/hulk_interference)
+	RegisterSignal(sparring, COMSIG_ATOM_ATTACK_HAND, .proc/hand_interference)
+	RegisterSignal(sparring, COMSIG_ATOM_ATTACK_PAW, .proc/paw_interference)
+	RegisterSignal(sparring, COMSIG_ATOM_HITBY, .proc/thrown_interference)
+	RegisterSignal(sparring, COMSIG_ATOM_BULLET_ACT, .proc/projectile_interference)
+	//severe flubs (insta match ender, no winners) conditions
+	RegisterSignal(sparring, COMSIG_LIVING_DEATH, .proc/death_flub)
+	RegisterSignal(sparring, COMSIG_PARENT_QDELETING, .proc/deletion_flub)
+
+/datum/sparring_match/proc/unhook_signals(mob/living/carbon/human/sparring)
+	if(!sparring)
+		return
+	UnregisterSignal(sparring, list(
+		COMSIG_MOB_FIRED_GUN,
+		COMSIG_MOB_GRENADE_ARMED,
+		COMSIG_MOB_ITEM_ATTACK,
+		COMSIG_MOVABLE_MOVED,
+		COMSIG_MOVABLE_TELEPORTED,
+		COMSIG_MOB_STATCHANGE,
+		COMSIG_PARENT_ATTACKBY,
+		COMSIG_ATOM_HULK_ATTACK,
+		COMSIG_ATOM_ATTACK_HAND,
+		COMSIG_ATOM_ATTACK_PAW,
+		COMSIG_ATOM_HITBY,
+		COMSIG_ATOM_BULLET_ACT,
+		COMSIG_LIVING_DEATH,
+		COMSIG_PARENT_QDELETING,
+	))
+
+///someone is changing health state, end the fight in crit
+/datum/sparring_match/proc/check_for_victory(datum/participant, new_stat)
+	SIGNAL_HANDLER
+
+	if(new_stat != SOFT_CRIT)
+		return
+	if(participant == chaplain)
+		end_match(opponent, chaplain)
+	else
+		end_match(chaplain, opponent)
+
+// SIGNALS THAT ARE FOR BEING ATTACKED FIRST (GUILTY)
+/datum/sparring_match/proc/outsider_interference(datum/source, obj/item/I, mob/attacker)
+	SIGNAL_HANDLER
+	if(attacker == chaplain || attacker == opponent)
+		// fist fighting a hulk is so dumb. i can't fathom why you would do this.
+		return
+	flub(attacker)
+
+/datum/sparring_match/proc/hulk_interference(datum/source, mob/attacker)
+	SIGNAL_HANDLER
+	if((attacker == chaplain || attacker == opponent))
+		// fist fighting a hulk is so dumb. i can't fathom why you would do this.
+		return
+	flub(attacker)
+
+/datum/sparring_match/proc/hand_interference(datum/source, mob/living/attacker)
+	SIGNAL_HANDLER
+	if(!attacker.combat_mode)
+		//not worth the flub
```

Comment:
I'd actually argue that this is worth the flub, as an outside party could, say, help shake a participant up to their feet or repeatedly try to grab them